server/controllers/api/videos/upload.ts

```diff
@@ -210,7 +216,12 @@ async function addVideo (options: {
 
   createTorrentFederate(video, videoFile)
     .then(() => {
-      if (video.state !== VideoState.TO_TRANSCODE) return
+      if (video.state !== VideoState.TO_TRANSCODE) {
+        // Video will be published before move is complete which may cause some video connections to drop
+        // But it's recommended to enable transcoding anyway, so this is the tradeoff
+        addMoveToObjectStorageJob(video, videoFile)
```

Comment:
Could we rename this function to `addMoveToObjectStorageJobIfNeeded`? I know it's a long name but when we read the code we know we did not forget the `if (CONFIG.OBJECT_STORAGE.ENABLED)`  here